backend/app/adapter/routing/handle/search.go

```diff
@@ -1,17 +1,97 @@
 package handle
 
 import (
+	"encoding/json"
+	"io/ioutil"
 	"net/http"
 
 	"github.com/short-d/app/fw/router"
+	"github.com/short-d/short/backend/app/usecase/authenticator"
 	"github.com/short-d/short/backend/app/usecase/search"
+	"github.com/short-d/short/backend/app/usecase/search/order"
 )
 
 // Search fetches resources under certain criterias.
 func Search(
-	search search.Search,
+	searcher search.Search,
+	authenticator authenticator.Authenticator,
 ) router.Handle {
 	return func(w http.ResponseWriter, r *http.Request, params router.Params) {
-		w.Write([]byte("not implemented"))
+		user := getUser(r, authenticator)
+		if user == nil {
+			w.Write([]byte("user not authenticated"))
+			return
+		}
+
+		searchRequest, err := readSearchRequest(r)
+		if err != nil {
+			http.Error(w, err.Error(), 500)
+			return
+		}
+
+		_ = search.Query{
+			Query: searchRequest.Query,
+			User:  user,
+		}
+		_, err = getFilter(searchRequest)
+		if err != nil {
+			http.Error(w, err.Error(), 500)
+			return
+		}
+
+		w.Write([]byte("request is read"))
+	}
+}
+
+// SearchRequest represents the JSON request received from Search API.
+type SearchRequest struct {
+	Query  string `json:"query"`
+	Filter Filter `json:"filter"`
+}
+
+// Filter represents the filter JSON field received from Search API.
+type Filter struct {
+	MaxResults int      `json:"max_results"`
+	Resources  []string `json:"resources"`
+	Orders     []string `json:"orders"`
+}
+
+func readSearchRequest(r *http.Request) (*SearchRequest, error) {
+	b, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	var searchRequest SearchRequest
+	err = json.Unmarshal(b, &searchRequest)
+	return &searchRequest, err
+}
```

Comment:
Please considering defining custom unmarshal function instead of letting API consumers invoke parse functions manually: https://gist.github.com/odeke-em/a2f0ad491c5d097e1ea29ecd036644a5